docs: everest mount page refresh

nopcoder opened the pull request:
- improved structure of mount documentation
- more information about how everest mount cache works

talSofer commented on docs/src/reference/mount.md:
```diff
@@ -9,968 +9,856 @@ status: enterprise
-### OS and Protocol Support
+- **Simplified Data Loading**: Use your existing tools to read and write files directly from the filesystem with no need for custom data loaders or SDKs.
+- **Seamless Scalability**: Scale from a few local files to billions without changing your tools or workflow. Use the same code from experimentation to production.
+- **Enhanced Performance**: Everest supports billions of files and offers fast, lazy data fetching, making it ideal for optimizing GPU utilization and other performance-sensitive tasks.
```
The use case title sounds like a technical benefit of mount. But there is a use case for performant data loading which I believe should be highlighted instead. WDYT?

talSofer commented on docs/src/reference/mount.md:
```diff
@@ -9,968 +9,856 @@ status: enterprise
+    After completing this getting started guide, we recommend reading the [Core Concepts](#core-concepts) section to understand caching, consistency, and performance characteristics.
 
-## Authenticate with lakeFS Credentials
+### 1. Prerequisites
```
nit; I would remove the numbers, it reduces clarity

talSofer commented on docs/src/reference/mount.md:
```diff
@@ -9,968 +9,856 @@ status: enterprise
-- Modifying file metadata (chmod, chown, chgrp, time) will result in noop (the file metadata will not be changed). 
-- When calling `remove` we mark a file as a tombstone using [Extended Attributes](https://en.wikipedia.org/wiki/Extended_file_attributes) API's.
-- Removal is not an atomic operation, calling remove and open at the same time might result in a race condition where the open might succeed.
+#### How Caching Works
```
Can we keep all heading 4 out of the toc?

talSofer commented on docs/src/reference/mount.md:
```diff
@@ -9,968 +9,856 @@ status: enterprise
+## Core Concepts
 
-#### Not Supported
+Before diving into data operations, it's important to understand how Everest manages performance, consistency, and caching. These concepts apply to both local and Kubernetes deployments.
```
```suggestion
This section will help you understand how Everest manages performance, consistency, and caching in both local and Kubernetes deployments.
```

talSofer commented on docs/src/reference/mount.md:
```diff
@@ -9,968 +9,856 @@ status: enterprise
+### Cache Behavior
 
-#### Behavior modified
+Everest uses a local cache to improve performance when accessing files from lakeFS. Understanding how the cache works will help you optimize performance for your specific use case.
```
I think that the guide should give explicit guidelines for how to optimize performance. This line makes the reader work harder to try and keep up and understand how to to it

nopcoder replied:
This section was added as part of questions to understand better how the cache works.

Your comment is for us adding another section about how to improve performance and keeping this section to understand how the cache works?

talSofer commented on docs/src/reference/mount.md:
```diff
@@ -9,968 +9,856 @@ status: enterprise
+### Consistency & Data Behavior
 
-### Commit Command (write-mode only)
+Understanding how Everest handles data consistency is crucial for working effectively with mounted lakeFS repositories.
```
I would remove this line

talSofer commented on docs/src/reference/mount.md:
```diff
@@ -9,968 +9,856 @@ status: enterprise
 
-When reading large files, Everest can fetch from lakeFS only the parts actually accessed.
-This can be useful for streaming workloads or for applications handling file formats such as Parquet, m4a, zip, tar that do not need to read the entire file.
+Everest is optimized for high-performance data access, but understanding these characteristics will help you get the best results:
```
```suggestion
Everest achieves high-performance data access through:
```

nopcoder commented on docs/src/reference/mount.md:
```diff
@@ -9,968 +9,856 @@ status: enterprise
+-   **Security Context:** Setting Pod `securityContext` (e.g., `runAsUser`) is not currently supported.
 
-**Helm Chart default values:**
+### 1. Prerequisites
```
Prerequisites at this level is part of the CSI driver

nopcoder commented on docs/src/reference/mount.md:
```diff
@@ -9,968 +9,856 @@ status: enterprise
-If `<mount_directory>` not specified, the command searches for the mount directory in the current working directory and upwards based on `.everest` directory existence.
-Please note that the diffs are from the source branch state at the time of mounting and not the current state of the source branch, i.e., changes to the source branch from other operations will not be reflected in the diff result.
+-   **Commit-Based Caching**: Each commit ID has its own cache namespace. This ensures that cached data always corresponds to the correct version of your files.
+-   **Cache Invalidation on Commit**: When you commit changes in write mode using `everest commit`, the mount point's source commit ID is updated to the new HEAD of the branch. As a result, the cache associated with the old commit ID is no longer used, and new data will be cached under the new commit ID.
```
sure - each release we will need to update the documentation to reflect the user whats running on their machine.

nopcoder commented:
@talSofer addressed some of the feedback, I would like to do it incremental and enable other updates for the Everest for Windows before address more layout changes.
